node/config.js

```diff
@@ -48,13 +64,171 @@ function getUserFromToken(authToken) {
     }
 };
 
+function getUserDeviceMap(user) {
+    var userDeviceMap = userDeviceMapping[user];
+    if (!userDeviceMap) {
+        userDeviceMap = {
+            devices: []
+        };
+        userDeviceMapping[user] = userDeviceMap;
+    }
+
+    // console.log("----------------------- getUserDeviceMap");
+    // console.log(userDeviceMapping[user]);
+    // console.log("-----------------------");
+
+    return userDeviceMapping[user];
+}
+
+function getUserHubDevice(user, hubId) {
+    var userDeviceMap = getUserDeviceMap(user);
+    if (!userDeviceMap[hubId]) {
+        userDeviceMap[hubId] = { 
+            id: undefined,
+            onboardingAnswers: [],
+            onboardingOutput: undefined,
+            translator: undefined,
+            devices: []
+        };
+    }
+
+    // console.log("----------------------- getUserHubDevice");
+    // console.log(userDeviceMap[hubId]);
+    // console.log("-----------------------");
+
+    return userDeviceMap[hubId];
+}
+
+function getUserDevice(user, deviceId) {
+    var userDeviceMap = getUserDeviceMap(user);
+    
+    // console.log("----------------------- getUserDevice");
+    // console.log(userDeviceMap);
+    // console.log("-----------------------");
+
+    for (var i = 0; i < userDeviceMap.devices.length; i++) {
+        var d = userDeviceMap.devices[i];
+        if (d.id == deviceId) {
+            return d;
+        }
+    }
+
+    return undefined;
+}
+
+function addUserDevice(user, device) { 
+    var userDeviceMap = getUserDeviceMap(user);
+
+    var internalDevice = {};
+    internalDevice.id = userDeviceMap.devices.length;
+    internalDevice.name = device.name;
+    internalDevice.actualDevice = device;
+
+    userDeviceMap.devices.push(internalDevice);    
+}
+
+function addUserDeviceToDevice(user, device, parentDeviceId) {
```

Comment:
This sounds very confusing. We should add comments to all these functions to describe what they do... is there a UserDevice and a Device? Is the former "my thermostat" and the latter "any thermostat"? Would be nice to document the object relationships a tiny bit for readability. Not a blocking issue, we can follow up with a commenting and cleanup commit.